ProcessLib/ThermoHydroMechanics/ThermoHydroMechanicsProcessData.h

```diff
@@ -0,0 +1,139 @@
+          solid_specific_heat_capacity(solid_specific_heat_capacity_),
+          solid_thermal_conductivity(solid_thermal_conductivity_),
+          fluid_specific_heat_capacity(fluid_specific_heat_capacity_),
+          fluid_thermal_conductivity(fluid_thermal_conductivity_),
+          reference_temperature(reference_temperature_),
+          specific_body_force(std::move(specific_body_force_))
+    // reference_temperature(reference_temperature_)
+    {
+    }
+
+    ThermoHydroMechanicsProcessData(ThermoHydroMechanicsProcessData&& other) =
+        default;
+
+    //! Copies are forbidden.
+    ThermoHydroMechanicsProcessData(ThermoHydroMechanicsProcessData const&) =
+        delete;
+
+    //! Assignments are not needed.
+    void operator=(ThermoHydroMechanicsProcessData const&) = delete;
+
+    //! Assignments are not needed.
+    void operator=(ThermoHydroMechanicsProcessData&&) = delete;
+
+    MeshLib::PropertyVector<int> const* const material_ids;
+
+    /// The constitutive relation for the mechanical part.
+    /// \note Linear elasticity is the only supported one in the moment.
+    std::map<
+        int,
+        std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>>
+        solid_materials;
+    /// Permeability of the solid. A scalar quantity, Parameter<double>.
+    Parameter<double> const& intrinsic_permeability;
+    /// Volumetric average specific storage of the solid and fluid phases.
+    /// A scalar quantity, Parameter<double>.
+    Parameter<double> const& specific_storage;
+    /// Fluid's viscosity. A scalar quantity, Parameter<double>.
+    Parameter<double> const& fluid_viscosity;
+    /// Fluid's density. A scalar quantity, Parameter<double>.
+    Parameter<double> const& fluid_density;
+    /// Biot coefficient. A scalar quantity, Parameter<double>.
+    Parameter<double> const& biot_coefficient;
+    /// Porosity of the solid. A scalar quantity, Parameter<double>.
+    Parameter<double> const& porosity;
+    /// Solid's density. A scalar quantity, Parameter<double>.
+    Parameter<double> const& solid_density;
+    Parameter<double> const& solid_linear_thermal_expansion_coefficient;
+    Parameter<double> const& fluid_volumetric_thermal_expansion_coefficient;
+    Parameter<double> const& solid_specific_heat_capacity;
+    Parameter<double> const& solid_thermal_conductivity;
+    Parameter<double> const& fluid_specific_heat_capacity;
+    Parameter<double> const& fluid_thermal_conductivity;
+    Parameter<double> const& reference_temperature;
+    /// Specific body forces applied to solid and fluid.
+    /// It is usually used to apply gravitational forces.
+    /// A vector of displacement dimension's length.
+    Eigen::Matrix<double, DisplacementDim, 1> const specific_body_force;
+    double dt = 0.0;
+    double t = 0.0;
+
+    // double const reference_temperature;
+
+    MeshLib::PropertyVector<double>* pressure_interpolated = nullptr;
```

Comment:
Please add temperature_interpolated as well.

Reply:
Corrected

Reply:
Of course, you have to fill the array with some values...

Reply:
Corrected. The function is called in FEM-impl.h.